components/brave_origin/policy/brave_origin_policy_provider.cc

```diff
@@ -0,0 +1,261 @@
+  bool is_brave_origin_user = IsBraveOriginEnabled();
+
+  if (!is_brave_origin_user) {
+    return bundle;
+  }
+
+  // Create policy map for Chrome domain
+  policy::PolicyMap& policy_map = bundle.Get(
+      policy::PolicyNamespace(policy::POLICY_DOMAIN_CHROME, std::string()));
+
+  // Set BraveOrigin policies using mappings from BraveOriginService
+  const auto policy_mappings = BraveOriginService::GetPolicyKeyMappings();
+
+  if (!local_state_) {
+    return bundle;
+  }
+
+  for (const auto& [policy_key, pref_name] : policy_mappings) {
+    const BraveOriginPrefInfo* pref_info =
+        BraveOriginService::GetPrefInfo(pref_name);
+    if (!pref_info) {
+      continue;
+    }
+    if (pref_info->scope == BraveOriginPolicyScope::kProfile) {
+      SetBraveOriginPolicyForPref(policy_map, policy_key, pref_name,
+                                  local_state_);
+    } else if (pref_info->scope == BraveOriginPolicyScope::kGlobal) {
+      SetBraveOriginGlobalPolicyForPref(policy_map, policy_key, pref_name,
+                                        local_state_);
+    } else {
+      continue;
+    }
+  }
+
+  return bundle;
+}
+
+bool BraveOriginPolicyProvider::IsPolicySetByExternalProvider(
+    const std::string& policy_key) const {
+  if (!local_state_) {
+    return false;
+  }
+
+  // Get the pref name for this policy key
+  const auto policy_mappings = BraveOriginService::GetPolicyKeyMappings();
+  auto it = policy_mappings.find(policy_key);
+  if (it == policy_mappings.end()) {
+    return false;
+  }
+
+  const std::string& pref_name = it->second;
+
+  // Build the externally managed key
+  std::string externally_managed_key =
+      pref_name + brave_origin::prefs::kExternallyManagedSuffix;
+
+  // Check local state for the external management flag
+  const base::Value::Dict& policies_dict =
+      local_state_->GetDict(prefs::kBravePolicies);
+  const base::Value* managed_value = policies_dict.Find(externally_managed_key);
+
+  return managed_value && managed_value->is_bool() && managed_value->GetBool();
+}
+
+void BraveOriginPolicyProvider::SetBraveOriginPolicyForPref(
+    policy::PolicyMap& policy_map,
+    const std::string& policy_key,
+    const std::string& pref_name,
+    PrefService* local_state) {
+  // Only set policy if not already set by external providers
+  bool is_external = IsPolicySetByExternalProvider(policy_key);
+
+  if (!is_external) {
+    // Get the brave_policies dictionary and iterate through all keys
+    const base::Value::Dict& policies_dict =
+        local_state->GetDict(brave_origin::prefs::kBravePolicies);
+
+    bool found_policy = false;
+    // Look for keys that end with our pref_name (format: profile_id.pref_name)
+    for (const auto item : policies_dict) {
+      const std::string& dict_key = item.first;
+      const base::Value& policy_value = item.second;
+
+      // Check if this key is for our pref_name
+      // Key format is: profile_id.pref_name
```

Comment:
What happens if there are conflicting Origin policy settings in multiple profiles? It looks like we just respect the setting of the first profile found, is that right?